mobile/src/components/uam-configure-group/components/permissions-table/permissions-table.tsx

```diff
@@ -18,25 +24,50 @@ type Props = {
     page: number;
     setPage: (page: number) => void;
   };
+  checkedIds: number[];
 };
 
 const PermissionsTable: FC<Props> = ({
   permissions,
   onCheckboxToggle,
   pagination,
+  checkedIds,
 }) => {
   const { control, reset } = useAppForm({ defaultValues: {} });
 
   const permissionRows = getPermissionsRows({
     permissions: permissions,
-    onToggle: onCheckboxToggle,
     control,
+    onToggle: onCheckboxToggle,
   });
   const permissionColumns = getPermissionsColumns();
 
+  const selectedItems: SelectedItemsValues = {};
+
+  useFocusEffect(
+    useCallback(() => {
+      reset(selectedItems);
+    }, [permissions, checkedIds]),
+  );
+
+  useEffect(() => {
+    checkedIds.map(
+      (id) =>
+        (selectedItems[`permissionIds.${id}`] = Boolean(
+          permissions.find((permission) => permission.id === id),
+        )),
+    );
+  }, [permissions, checkedIds]);
+
+  useEffect(() => {
+    reset(selectedItems);
+  }, [permissions]);
```

Comment:
Can't we replace all these useEffect/useFocusEffect with this?
```suggestion
  useFocusEffect(
    useCallback(() => {
      const selectedItems = // use reduce instead of map to get selected items
      reset(selectedItems);
    }, [permissions, checkedIds]),
  );
```

Reply:
Every day, I continue learning) Thanks!